src/main/java/hudson/plugins/emailext/EmailExtStep.java

```diff
@@ -0,0 +1,191 @@
+package hudson.plugins.emailext;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.inject.Inject;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.plugins.emailext.plugins.EmailTrigger;
+import hudson.plugins.emailext.plugins.trigger.AlwaysTrigger;
+import jenkins.model.Jenkins;
+import hudson.Util;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.CheckForNull;
+import javax.management.Descriptor;
+
+/**
+ * Created by acearl on 9/14/2015.
+ */
+
+public class EmailExtStep extends AbstractStepImpl {
+
+    public final String subject;
+
+    public final String body;
+
+    @CheckForNull
+    private String attachmentsPattern;
+
+    @CheckForNull
+    private String to;
+
+    @CheckForNull
+    private String replyTo;
+
+    @CheckForNull
+    private String mimeType;
+
+    private boolean attachLog;
+
+    private boolean compressLog;
+
+    @DataBoundConstructor
+    public EmailExtStep(String subject, String body) {
+        this.subject = subject;
+        this.body = body;
+    }
+
+    public @CheckForNull String getAttachmentsPattern() {
+        return attachmentsPattern == null ? "" : attachmentsPattern;

```

Comment:
Can it be `null`? In `setAttachmentsPattern` method you are validating the input.